src/VirtoCommerce.NotificationsModule.MicrosoftGraph/MicrosoftGraphEmailNotificationMessageSender.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Microsoft.Extensions.Options;
+using Microsoft.Graph;
+using Microsoft.Graph.Models;
+using Microsoft.Graph.Users.Item.SendMail;
+using VirtoCommerce.NotificationsModule.Core.Exceptions;
+using VirtoCommerce.NotificationsModule.Core.Model;
+using VirtoCommerce.NotificationsModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.NotificationsModule.MicrosoftGraph;
+
+public class MicrosoftGraphEmailNotificationMessageSender(
+    IOptions<MicrosoftGraphSenderOptions> microsoftGraphOptions,
+    IOptions<EmailSendingOptions> emailSendingOptions)
+    : INotificationMessageSender
+{
+    public const string Name = "MicrosoftGraph";
+
+    private readonly MicrosoftGraphSenderOptions _microsoftGraphOptions = microsoftGraphOptions.Value;
+    private readonly EmailSendingOptions _emailSendingOptions = emailSendingOptions.Value;
+
+    public virtual bool CanSend(NotificationMessage message) => message is EmailNotificationMessage;
+
+    public async Task SendNotificationAsync(NotificationMessage message)
+    {
+        var emailNotificationMessage = message as EmailNotificationMessage
+            ?? throw new ArgumentException($"The message is not {nameof(EmailNotificationMessage)} type");
+
+        try
+        {
+            var graphMessage = new Message
+            {
+                Subject = emailNotificationMessage.Subject,
+                Body = new ItemBody { ContentType = BodyType.Html, Content = emailNotificationMessage.Body },
+                From = NewRecipient(emailNotificationMessage.From ?? _emailSendingOptions.DefaultSender),
+                ToRecipients = [NewRecipient(emailNotificationMessage.To)]
+            };
+
+            if (!emailNotificationMessage.CC.IsNullOrEmpty())
+            {
+                graphMessage.CcRecipients ??= [];
+                foreach (var ccEmail in emailNotificationMessage.CC)
+                {
+                    graphMessage.CcRecipients.Add(NewRecipient(ccEmail));
+                }
+            }
+
+            if (!emailNotificationMessage.BCC.IsNullOrEmpty())
```

Comment:
The same idea. This block can be replaced with a single line
graphMessage.BccRecipients = emailNotificationMessage.BCC.Select(NewRecipient).ToList();